config.ru

```diff
@@ -4,16 +4,39 @@ require 'routemaster/receiver'
 require 'dotenv'
 require 'sinatra'
 require 'pry'
+require 'data_sink_client'
 
 Dotenv.load!
 
+SUBSCRIBER_NAME = 'routemaster-blackhole'
+
 class Handler
   def on_events(events)
     events.each do |event|
       event.keys.each { |k| event[k.to_sym] = event.delete(k) }
       $stderr.write("%<topic>-12s %<type>-12s %<url>s %<t>s\n" % event)
       $stderr.flush
     end
+    send_to_data_sink(events)
+  end
+
+  private
+
+  def send_to_data_sink(events)
+    return unless data_sink_enabled?
+    events.map do |event|
+      Thread.new do
```

Comment:
We should use a thread-pool to ensure we never try to create a thread per event for large batches.

I know the `parallel` gem would make this really easy, but maybe we already have a favourite thread-pool gem.

Reply:
@mjkillough but here we are creating a thread per event. Are you worried about spawning too much threads per process?

Reply:
Yeah.

The default batch size for Routemaster is 100, which is more threads than I'd normally start in one go. I would prefer to limit the number of threads we create, but I don't think it'd really be disastrous to leave it as-is. This service doesn't do anything business critical.

Reply:
https://github.com/ruby-concurrency/concurrent-ruby/blob/master/doc/thread_pools.md is also nice (we're using it in data-sink)